src/sentry/snuba/discover.py

```diff
@@ -466,9 +466,9 @@ def top_events_timeseries(
         )
 
         for field in selected_columns:
-            # project is handled by filter_keys already
+            # If we have a project field, we need to limit results by project so we dont hit the result limit
             if field in ["project", "project.id"]:
```

Comment:
in selected columns the field will be `project` (project.id added here for safety basically), and project.id is added to the results automatically because of the project slug transformation